lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -1681,7 +1681,7 @@ function computeLegendForLayer(catalogItem, thisLayer, styleName) {
     if (!defined(legendUri)) {
         // Construct a GetLegendGraphic request.
         legendUri = new URI(cleanUrl(catalogItem.url) +
-            '?service=WMS&version=1.1.0&request=GetLegendGraphic&format=image/png&transparent=True&layer=' + encodeURIComponent(thisLayer.Name));
+            '?service=WMS&version=1.1.0&request=GetLegendGraphic&format=image/png&transparent=True&layer=' + encodeURIComponent(thisLayer.Name) + '&styles=' + encodeURIComponent(style.Name));
```

Comment:
Need to handle the possibility that `style` is undefined.

Reply:
I think we still want to generate a legend even if `style` is undefined, we just won't include the `style=` parameter.

Reply:
Have added `if (!defined(legendUri) && defined(style) && defined(style.Name)) {` similar to what is done above

Reply:
But now if there's no `style`, there won't be a `legendUri` at all, right?

Reply:
Ok yeah I see... so we just pass an empty string in that case? eg
````
if (!defined(legendUri)) {
   const styleToUse = defined(style) && defined(style.Name) ? style.Name : '';
````

Reply:
It'd be slightly better to leave out the `style=` entirely I think.